MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m

```diff
@@ -148,8 +189,100 @@ - (MSPriority)priority {
   return MSPriorityHigh;
 }
 
-- (MSInitializationPriority)initializationPriority {
-  return MSInitializationPriorityDefault;
+#pragma mark - Private
+
+- (BOOL)checkURLSchemeRegistered:(NSString *)urlScheme {
+  NSArray *schemes;
+  NSArray *types = [MS_APP_MAIN_BUNDLE objectForInfoDictionaryKey:@"CFBundleURLTypes"];
+  for (NSDictionary *urlType in types) {
+    schemes = urlType[@"CFBundleURLSchemes"];
+    for (NSString *scheme in schemes) {
+      if ([scheme isEqualToString:urlScheme]) {
+        return YES;
+      }
+    }
+  }
+  return NO;
+}
+
+- (NSURL *)buildTokenRequestURLWithAppSecret:(NSString *)appSecret error:(NSError *__autoreleasing *)error {
+
+  // Compute URL path string.
+  NSString *urlPath = [NSString stringWithFormat:kMSUpdtsUpdateTokenApiPathFormat, appSecret];
+
+  // Build URL string.
+  NSString *urlString = [kMSDefaultInstallUrl stringByAppendingString:urlPath];
+  NSURLComponents *components = [NSURLComponents componentsWithString:urlString];
+
+  // Check URL validity so far.
+  if (!components && error) {
```

Comment:
You should validate even though error is `nil`. `error` should be tested in the `if`.